docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+In the current design of ccv1, the following two types of transport are supported:
+
+- "docker": used to match the image in the container registry that implements the docker registry HTTP API V2 protocol.
+
+- "dir": used to match images located in the local file system.
+
+For the "docker" transport type, the scope can be a docker reference named in a fully expanded form, and tag or digest can be additionally specified for exact matching, for example: `docker.io/library/busybox:latest`. Generic matching is also supported:
+
+- repository namespace: for example, `docker.io/openshift`, which matches the images as `openshift/hello-openshift` and `openshift/origin`.
+
+- repository host: for example, `docker.io`, matches all images from the docker.io repository.
+
+- subdomain name matching: such as `*.temporary-project.example.com`. Note: wildcards can only appear in the top sub domain name. Wildcards like `example.*.com` are not allowed to be specified.
+
+For the "dir" transport type, the scope format is the directory path. The path must be absolute and contain no symbolic links. The default scope is "".
+
+The policy file must be generated and maintained by the user. Additional security means need to be used to ensure that the policy file will not be modified by malicious entities during storage and transmission. If users have high requirements for security level, they even need to use additional means to ensure the confidentiality of policy files.
+
+If image-rs finds that there is no policy.json in the /etc/containers/ directory or If the policy.json file is empty, call the GetResource grpc service provided by the attestation-agent. After remote attestation, obtain it from the user controlled relying KBS through the encrypted channel.
+
+The format of the policy.json file follows the format and syntax defined in the [container/image policy.json](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) document.
+
+In the V1 design of the confidential containers, a safe and reasonable policy.json file may look like the following example: 
+
+```json
+{
+    "default": [{"type": "reject"}], 
+    "transports": {
+        "docker": {
+            "docker.io/my_private_registry": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyData": "<public Key>",
+                }
+            ],
+            "registry.access.redhat.com": [
+                {
+                    "type": "signedBy",
+                    "keyType": "GPGKeys",
+                    "keyPath": "/etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release",
+                }
+            ],
+        },
+        "dir": {
+            "": [{"type": "insecureAcceptAnything"}]
+        }
+    }
+}
+```
+
+Before pulling the container image layers, match the policy in the following order according to the location/registry of the container image and execute the corresponding policies:
```

Comment:
It's useful to have this information here, but I wonder if this policy matching order is not specified somewhere else?